File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        first, second = 0, 1
        maxLen = min(1, len(s))
        counter = maxLen

        while second < len(s):
            if s[second] in s[first:second]:
                maxLen = max(maxLen, counter)
                while first <= second and s[second] in s[first:second]:
                    first += 1
                second = second + 1
                counter = max(0, second - first)
            else:
                second += 1
                counter += 1

        return max(maxLen, counter)
                

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-3.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int len = nums.length;

        int first = 0;
        int second = first + 1;

        while (second < len) {
            // find first not matching val
            while (second < len && nums[second] == nums[first]) second++;

            if (second < len) {
                first++;
                nums[first] = nums[second];
                second++;
            }
        }

        return first + 1;
        
    }
}